File: app/Exceptions/IrailHttpException.php
<?php

namespace Irail\Exceptions;

use Symfony\Component\HttpKernel\Exception\HttpException;

class IrailHttpException extends HttpException
{
    public function __construct(int $statusCode, string $message = '', \Throwable $previous = null, array $headers = [])
    {
        parent::__construct($statusCode, $message, $previous, $headers, $statusCode);
    }
}

File: src/main/java/be/irail/api/exception/IrailHttpException.java
package be.irail.api.exception;

public class IrailHttpException extends RuntimeException {
    private final int httpCode;

    public IrailHttpException(int httpCode, String message) {
        super(message);
        this.httpCode = httpCode;
    }

    public IrailHttpException(int httpCode, String message, Throwable cause) {
        super(message, cause);
        this.httpCode = httpCode;
    }

    public int getHttpCode() {
        return httpCode;
    }
}
